### PR TITLE
*: Address markdown linting violations in various documentation files

### PR DESCRIPTION
**Pull request by timflannagan**
Update the context/* documentation files:

- Add code formatting blocks.
- Normalize punctuation in lists.
- Add newline between headings and text.
- Add newline between text and first list element.
- Add missing trailing newlines.
- Remove any redundant newlines.
- Remove any redundant spaces.

### CODE REVIEW

**Comment by anik120** [comment]
@timflannagan1 our CI is failing on a [few links](https://travis-ci.org/github/operator-framework/olm-docs/builds/756636495#L254-L269) in the site. Is it possible to fix those links as a part of this PR? Thanks!

**Comment by timflannagan1** [comment]
> @timflannagan1 our CI is failing on a [few links](https://travis-ci.org/github/operator-framework/olm-docs/builds/756636495#L254-L269) in the site. Is it possible to fix those links as a part of this PR? Thanks!

Yeah, that's fine with me. I'll play around with running this locally.

File: hack/ci/link-check.sh [hunk 1]
```diff
@@ -2,6 +2,6 @@
 set -ev
 
 docker volume create olm-html
-docker run --rm -v "$(pwd):/src" -v olm-html:/src/public klakegg/hugo:0.73.0-ext-ubuntu
+docker run --rm --privileged=true -v "$(pwd):/src" -v olm-html:/src/public:Z klakegg/hugo:0.73.0-ext-ubuntu
```
Comment (timflannagan1) [inline hack/ci/link-check.sh:5]: @anik120 It looks like we need to update the README instructions for running this check locally for boxes that have these filesystems owned by root.

You likely wouldn't want to run this as a privileged container but documenting ways to overcome any potential permission errors when mounting that volume would be a step in the right direction.

Comment (timflannagan1) [reply hack/ci/link-check.sh:5]: Pulled out those changes into their own PR. I'll continue messing around with fixing those external/internal link errors that the linting script is producing.

Comment (timflannagan1) [reply hack/ci/link-check.sh:5]: Ended up pulling out the changes needed to get CI green again in #101.

**Comment by timflannagan** [comment]
/retitle *: Address markdown linting violations in various documentation files

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
It looks like I never got around to breaking that WIP/TODO commit into multiple commits, but it also looks like multiple people are trying to contribute to this repository at the same time. In order to make sure others don't have to rebase constantly, I'm going to merge this without fixing that commit message.

**Pull request merged**